src/plugins/apimon/apimon.cpp

```diff
@@ -0,0 +1,390 @@
+            break;
+        case OUTPUT_JSON:
+            escaped_pname = drakvuf_escape_str(info->proc_data.name);
+            printf( "{"
+                    "\"Plugin\": \"apimon\", "
+                    "\"TimeStamp\":" "\"" FORMAT_TIMEVAL "\", "
+                    "\"ProcessName\": %s, "
+                    "\"UserName\": \"%s\", "
+                    "\"UserId\": %" PRIu64 ", "
+                    "\"PID\": %d, "
+                    "\"PPID\": %d, "
+                    "\"Method\": \"%s\", "
+                    "\"CalledFrom\": 0x%" PRIx64 ", "
+                    "\"ReturnValue\": 0x%" PRIx64 ", "
+                    "\"Arguments\": ",
+                    UNPACK_TIMEVAL(info->timestamp),
+                    escaped_pname,
+                    USERIDSTR(drakvuf), info->proc_data.userid,
+                    info->proc_data.pid, info->proc_data.ppid,
+                    info->trap->name,
+                    info->regs->rip,
+                    info->regs->rax);
+
+            print_arguments(drakvuf, info, ret_target->arguments, ret_target->argument_printers);
+            printf(", "
+                   "\"Extra\": {");
+            print_extra_data(extra_data);
+            printf("}}");
+            g_free(escaped_pname);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[APIMON-USERHOOK] TIME:" FORMAT_TIMEVAL " VCPU:%" PRIu32 " CR3:0x%" PRIx64 " ProcessName:\"%s\" UserID:%" PRIi64 " Method:\"%s\" CalledFrom:0x%" PRIx64 " ReturnValue:0x%" PRIx64 " Arguments:",
+            UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+            info->proc_data.userid, info->trap->name, info->regs->rax, info->regs->rip);
+            print_arguments(drakvuf, info, ret_target->arguments, ret_target->argument_printers);
+            break;
+    }
+    printf("\n");
+
+    drakvuf_remove_trap(drakvuf, info->trap, nullptr);
+    delete ret_target;
+    return VMI_EVENT_RESPONSE_NONE;
+}
+
+static event_response_t usermode_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info* info)
+{
+    hook_target_entry_t* target = (hook_target_entry_t*)info->trap->data;
+
+    if (target->pid != info->proc_data.pid)
+        return VMI_EVENT_RESPONSE_NONE;
+
+    vmi_lock_guard lg(drakvuf);
+    vmi_v2pcache_flush(lg.vmi, info->regs->cr3);
+
+    bool is_syswow = drakvuf_is_wow64(drakvuf, info);
+
+    access_context_t ctx =
+            {
+                    .translate_mechanism = VMI_TM_PROCESS_DTB,
+                    .dtb = info->regs->cr3,
+                    .addr = info->regs->rsp
+            };
+
+    bool success = false;
+    addr_t ret_addr = 0;
+
+    if (is_syswow)
+    {
+        uint32_t ret_addr_tmp;
+
+        if (vmi_read_32(lg.vmi, &ctx, &ret_addr_tmp) == VMI_SUCCESS)
+        {
+            success = true;
+            ret_addr = ret_addr_tmp;
+        }
+    }
+    else
+    {
+        success = vmi_read_64(lg.vmi, &ctx, &ret_addr) == VMI_SUCCESS;
```

Comment:
Although I would wrap the right-hand side of this as `success = (...)` even if its not strictly necessary. I think it would help with readability.